platforms/software/testing-base-infrastructure/src/main/java/org/gradle/api/internal/tasks/testing/TestSources.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.testing;
+
+import org.gradle.api.tasks.testing.ClassSource;
+import org.gradle.api.tasks.testing.ClasspathResourceSource;
+import org.gradle.api.tasks.testing.CompositeTestSource;
+import org.gradle.api.tasks.testing.DirectorySource;
+import org.gradle.api.tasks.testing.FilePosition;
+import org.gradle.api.tasks.testing.FileSource;
+import org.gradle.api.tasks.testing.MethodSource;
+import org.gradle.api.tasks.testing.MissingSource;
+import org.gradle.api.tasks.testing.PackageSource;
+import org.gradle.api.tasks.testing.TestSource;
+import org.gradle.api.tasks.testing.UnknownSource;
+import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
+
+import java.io.File;
+import java.util.List;
+
+@NullMarked
+public class TestSources {
```

Comment:
💭 This type doesn't seem to be necessary.  I suppose it's nice to create all the different types of sources through this one factory type.  Could be `final` and should have javadoc.